code/controllers/subsystem/polls.dm

```diff
@@ -0,0 +1,381 @@
+			voted_ids += vote.player_id
+
+		var/datum/poll/active_poll = active_polls[id]
+
+		for(var/client/client as anything in GLOB.clients)
+			if(client.player_data.id in voted_ids)
+				continue
+
+			to_send_at_lobby += CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(to_chat), client, SPAN_LARGE("You have not voted in the '[active_poll.question]' poll. Click <a href='byond://?src=\ref[client.mob];poll=1'>here</a> to vote."))
+
+	RegisterSignal(SSdcs, COMSIG_GLOB_MODE_PREGAME_LOBBY, PROC_REF(handle_lobby))
+	RegisterSignal(SSdcs, COMSIG_GLOB_CLIENT_LOGGED_IN, PROC_REF(handle_new_user))
+
+	return SS_INIT_SUCCESS
+
+/// Clear and populate the poll lists with fresh data
+/datum/controller/subsystem/polls/proc/setup_polls()
+	active_polls = list()
+
+	var/list/datum/view_record/poll/all_active_polls = DB_VIEW(
+		/datum/view_record/poll,
+		DB_AND(
+			DB_COMP("active", DB_EQUALS, TRUE),
+			DB_COMP("expiry", DB_GREATER_EQUAL, time2text(world.realtime, "YYYY-MM-DD hh:mm:ss"))
+		)
+	)
+
+	for(var/datum/view_record/poll/poll as anything in all_active_polls)
+		var/datum/poll/new_poll = new(poll.question)
+
+		var/list/datum/view_record/poll_answer/poll_answers = DB_VIEW(
+			/datum/view_record/poll_answer,
+			DB_COMP("poll_id", DB_EQUALS, poll.id)
+		)
+
+		for(var/datum/view_record/poll_answer/answer as anything in poll_answers)
+			new_poll.answers["[answer.id]"] = answer.answer
+
+		active_polls["[poll.id]"] = new_poll
+
+	concluded_polls = list()
+
+	var/list/datum/view_record/poll/all_concluded_polls = DB_VIEW(
+		/datum/view_record/poll,
+		DB_AND(
+			DB_COMP("active", DB_EQUALS, TRUE),
+			DB_COMP("expiry", DB_LESS, time2text(world.realtime, "YYYY-MM-DD hh:mm:ss"))
+		)
+	)
+
+	for(var/datum/view_record/poll/poll as anything in all_concluded_polls)
+		var/datum/poll/concluded/concluded_poll = new(poll.question)
+
+		var/list/datum/view_record/poll_answer/poll_answers = DB_VIEW(
+			/datum/view_record/poll_answer,
+			DB_COMP("poll_id", DB_EQUALS, poll.id)
+		)
+
+		for(var/datum/view_record/poll_answer/answer as anything in poll_answers)
+			var/total = length(DB_VIEW(/datum/view_record/player_poll_vote, DB_COMP("answer_id", DB_EQUALS, answer.id)))
+
+			concluded_poll.answer_totals[answer.answer] = total
+
+		concluded_polls["[poll.id]"] = concluded_poll
+
+	update_static_data_for_all_viewers()
+
+/// Sends all the queued messages when we enter the
+/// lobby mode, so the messages do not get lost
+/datum/controller/subsystem/polls/proc/handle_lobby(source)
+	SIGNAL_HANDLER
+
+	for(var/datum/callback/callback as anything in to_send_at_lobby)
+		callback.InvokeAsync()
+
+/datum/controller/subsystem/polls/proc/handle_new_user(source, client/new_client)
+	SIGNAL_HANDLER
+
+	INVOKE_ASYNC(src, PROC_REF(remind_new_user), new_client)
+
+/// Reminds new users logging in of any uncompleted polls.
+/datum/controller/subsystem/polls/proc/remind_new_user(client/new_client)
+	set waitfor = FALSE
+
+	while(!new_client.player_data)
+		stoplag()
```

Comment:
UNTIL(new_client.player_data)

but honestly it could be useful to just strap signaling in load_player_data_info